### PR TITLE
add verify check to jobset to check that configs were committed

### PR DESCRIPTION
**Pull request by kannon92**
https://github.com/kubernetes-sigs/jobset/pull/90 is to add a verify check to jobset repo.  

This adds it into the CI to gate against api changes without the correspond code.

### CODE REVIEW

**Comment by kannon92** [comment]
@ahg-g or @danielvegamyhre could use your review here?

**Comment by ahg-g** [comment]
/hold

For Daniel's question

File: config/jobs/kubernetes-sigs/jobset/jobset-presubmit.yaml [hunk 1]
```diff
@@ -108,3 +108,20 @@ presubmits:
...
+    skip_if_only_changed: "^docs/|^\\.github/|\\.(md)$|^(README|LICENSE|OWNERS)$"
+    decorate: true
+    path_alias: sigs.k8s.io/jobset
+    annotations:
+      testgrid-dashboards: sig-apps
+      testgrid-tab-name: pull-jobset-verify-main
+      description: "Run jobset verify checks"
+    spec:
+      containers:
+      - image: golang:1.20
```
[the remark this answers is not in the document]
Comment (kannon92) [reply config/jobs/kubernetes-sigs/jobset/jobset-presubmit.yaml:123]: Yea that’s a good call.  But should we update the other presubmits to 1.20?

[the remark this answers is not in the document]
Comment (danielvegamyhre) [reply config/jobs/kubernetes-sigs/jobset/jobset-presubmit.yaml:123]: Sure I think that would be best

[the remark this answers is not in the document]
Comment (kannon92) [reply config/jobs/kubernetes-sigs/jobset/jobset-presubmit.yaml:123]: I’ll update that tomorrow!

[the remark this answers is not in the document]
Comment (kannon92) [reply config/jobs/kubernetes-sigs/jobset/jobset-presubmit.yaml:123]: updated.

**Pull request merged**